src/storage/btree.c

```diff
@@ -18704,8 +18704,20 @@ btree_compare_key (DB_VALUE * key1, DB_VALUE * key2, TP_DOMAIN * key_domain, int
 	  return DB_UNK;
 	}
 
-      if (TP_ARE_COMPARABLE_KEY_TYPES (key1_type, key2_type) && TP_ARE_COMPARABLE_KEY_TYPES (key1_type, dom_type)
-	  && TP_ARE_COMPARABLE_KEY_TYPES (key2_type, dom_type))
+      bool are_types_comparable = TP_ARE_COMPARABLE_KEY_TYPES (key1_type, key2_type)
+	&& TP_ARE_COMPARABLE_KEY_TYPES (key1_type, dom_type) && TP_ARE_COMPARABLE_KEY_TYPES (key2_type, dom_type);
+      if (are_types_comparable)
+	{
+	  // check strings codeset
+	  if (TP_IS_STRING_TYPE (key1_type) && TP_IS_STRING_TYPE (key2_type)
+	      && db_get_string_codeset (key1) != db_get_string_codeset (key2))
```

Comment:
I think we should compare collations : 
db_get_string_collation (key1) != db_get_string_collation (key2)

At this point, collations should be the same (if type_checking has done its job).
Then when entering tp_value_compare_with_error, if codesets are different there will be an error raised.

Reply:
Please clarify your suggestion - should codeset check be replaced with collation check or should both be checked?

Reply:
codeset check should be replaced with collation check

Reply:
replaced codeset check with collation check